makefile.gen

```diff
@@ -94,94 +115,104 @@ Debug: debug
 Release: release
 Asm: asm
 
-.PHONY: clean
-
 cleantmp:
 	$(RM) -f $(RES_RS)
+.PHONY: cleantmp
 
 cleandep:
 	$(RM) -f $(DEPS)
+.PHONY: cleandep
 
 cleanlst:
 	$(RM) -f $(LSTS)
+.PHONY: cleanlst
 
 cleanres: cleantmp
 	$(RM) -f $(RES_H) $(RES_DEP) $(RES_DEPS)
+.PHONY: cleanres
 
 cleanobj:
-	$(RM) -f $(OBJS) out/sega.o out/rom_head.bin out/rom_head.o out/rom.out
+	$(RM) -f $(OBJS) $(OUT)/sega.o $(OUT)/rom_head.bin $(OUT)/rom_head.o $(OUT)/rom.out
+.PHONY: cleanobj
 
 clean: cleanobj cleanres cleanlst cleandep
-	$(RM) -f out.lst out/cmd_ out/symbol.txt out/rom.nm out/rom.wch out/rom.bin
+	$(RM) -f out.lst $(OUT)/cmd_ $(OUT)/symbol.txt $(OUT)/rom.nm $(OUT)/rom.wch $(OUT)/rom.bin
+.PHONY: clean
 
 cleanrelease: clean
+.PHONY: cleanrelease
 
 cleandebug: clean
+.PHONY: cleandebug
 
 cleanasm: cleanlst
+.PHONY: cleanasm
 
 cleandefault: clean
 cleanDefault: clean
+.PHONY: cleandefault cleanDefault
 
 cleanRelease: cleanrelease
 cleanDebug: cleandebug
 cleanAsm: cleanasm
-
-pre-build:
-	$(MKDIR) -p $(SRC)/boot
-	$(MKDIR) -p out
+.PHONY: cleanRelease cleanDebug cleanAsm
 
 
-out/rom.bin: out/rom.out
-	$(OBJCPY) -O binary out/rom.out out/rom.bin
-	$(SIZEBND) out/rom.bin -sizealign 131072 -checksum
+$(OUT)/rom.bin: $(OUT)/rom.out
+	$(OBJCPY) -O binary $(OUT)/rom.out $(OUT)/rom.bin
+	$(SIZEBND) $(OUT)/rom.bin -sizealign 131072 -checksum
 
-out/symbol.txt: out/rom.out
-	$(NM) $(LTO_PLUGIN) -n out/rom.out > out/symbol.txt
+$(OUT)/symbol.txt: $(OUT)/rom.out
+	$(NM) $(LTO_PLUGIN) -n $(OUT)/rom.out > $(OUT)/symbol.txt
 
-out/rom.out: out/sega.o out/cmd_ $(LIBMD)
-	$(CC) -m68000 -B$(BIN) -n -T $(GDK)/md.ld -nostdlib out/sega.o @out/cmd_ $(LIBMD) $(LIBGCC) -o out/rom.out -Wl,--gc-sections -flto
-	$(RM) out/cmd_
+$(OUT)/rom.out: $(OUT)/sega.o $(OUT)/cmd_ $(LIBMD)
+	$(MKDIR) -p $(dir $@)
+	$(CC) -m68000 -B$(BIN) -n -T $(GDK)/md.ld -nostdlib $(OUT)/sega.o @$(OUT)/cmd_ $(LIBMD) $(LIBGCC) -o $(OUT)/rom.out -Wl,--gc-sections -flto
+	$(RM) $(OUT)/cmd_
 
-out/cmd_: $(OBJS)
-	$(ECHO) "$(OBJS)" > out/cmd_
+$(OUT)/cmd_: $(OBJS)
+	$(MKDIR) -p $(dir $@)
+	$(ECHO) "$(OBJS)" > $(OUT)/cmd_
 
-out/sega.o: $(SRC)/boot/sega.s out/rom_head.bin
-	$(CC) -x assembler-with-cpp -Wa,--register-prefix-optional,--bitwise-or $(AFLAGS) -c $(SRC)/boot/sega.s -o $@
+$(OUT)/sega.o: $(OUT)/boot/sega.s $(OUT)/rom_head.bin

```

Comment:
This doesn't actually seem to depend on rom_head.bin, so I'm removing that dependency.

Reply:
I stand corrected.  sega.s references rom_head.bin.  I didn't realize.  I'll add a comment.

Reply:
Done.